### ISSUE TITLE
[lldb] warning: using the result of an assignment as a condition without parentheses [-Wparentheses]

### ISSUE THREAD

**Issue by trcrsired** [llvm/llvm-project#107551]
```
50:29: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
  950 |   if (debugserver_file_spec = GetDebugserverPath(platform)) {
      |       ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/cqwrteur/toolchains_build/llvm-project/lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp:950:29: note: place parentheses around the assignment to silence this warning
  950 |   if (debugserver_file_spec = GetDebugserverPath(platform)) {
      |                             ^                             
      |       (                                                   )
/home/cqwrteur/toolchains_build/llvm-project/lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp:950:29: note: use '==' to turn this assignment into an equality comparison
  950 |   if (debugserver_file_spec = GetDebugserverPath(platform)) {
      |                             ^
      |                             ==
1 warning generated.
```

**Comment by trcrsired** [comment]
fixed: https://github.com/llvm/llvm-project/commit/d4e320e6f4dd78929cec44b342b6035a6bbd6323
